nixos/modules/services/misc/tiddlywiki.nix

```diff
@@ -0,0 +1,82 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.tiddlywiki;
+  listenParams = concatStrings (mapAttrsToList (n: v: " '${n}=${toString v}' ") cfg.listenOptions);
+  exe = "${cfg.package}/lib/node_modules/.bin/tiddlywiki";
+
+in {
+
+  options.services.tiddlywiki = {
+
+    enable = mkEnableOption "TiddlyWiki nodejs server";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.nodePackages.tiddlywiki;
+      defaultText = "pkgs.nodePackages.tiddlywiki";
+      description = "TiddlyWiki package to use.";
+    };
+
+    wikipath = mkOption {
+      type = types.path;
+      default = "/var/lib/tiddlywiki";
+      description = "Path to wiki folder.";
+    };
+
+    listenOptions = mkOption {
+      type = types.attrs;
+      default = {};
+      example = {
+        username = "RJW7RG9qbc8i9yhTVgqT";
+        password = "rS8Y4LSgxJKxdhU9hvod";
```

Comment:
Is there a way to specify passwords in a way that they won't be stored in the world readable nix store?

Reply:
good point, made some investigation in tiddlywiki auth, looks like user can pass csv file with credentials. I've updated example configuration and test

Reply:
Awesome!